### PR TITLE
Landing page

### PR DESCRIPTION
**Pull request by guy-roberts**
### Context

The landing page (also called hub page or seven areas of learning) is now available at /content

### Changes proposed in this pull request
Page titles can now include commas and full stops. Both are ignored in the slug

### Guidance to review
Recreate the database, migrate and seed.  Start rails locally, have a look for the landing page at https://bla.gov.uk/content

Navigate into a page,  check that the desktop menu matches the directory structure

Click on top links in the title and make sure they go tot he right place

### CODE REVIEW

File: app/models/content_page.rb [hunk 1]
```diff
@@ -16,7 +16,11 @@ class ContentPage < ApplicationRecord
 
   before_save :set_slug_from_title
 
+  # Infer the slug from the title
+  # Spaces in the title are replaced by a hypen
+  # hypens are left unchanged
+  # commas are not translated into the slug
   def set_slug_from_title
-    self.slug = title.downcase.gsub(/ /, "-")
+    self.slug = title.downcase.gsub(/ /, "-").gsub(",", "").gsub(".", "")
```
Comment (marcrice12) [inline app/models/content_page.rb:24]: .gsub(/ /, "-").gsub(/\,|\./, "")

File: app/webpacker/styles/_cards.scss [hunk 2]
```diff
@@ -0,0 +1,154 @@
+//styles for the NHScard component
+
+$card-border-width: 1px;
+$card-border-bottom-width: 3px;
+$card-border-color: #1d70b8;
+$card-border-hover-color: #003078;
+
+.nhsuk-card {
```
Comment (tonyto) [inline app/webpacker/styles/_cards.scss:8]: I can't find any usage of this class - why is it required? Can it be removed?

File: app/views/content_pages/index.html.erb [hunk 3]
```diff
@@ -1,18 +1,14 @@
...
-
   <% @content_pages.each_with_index do |content_page, index| %>
 
     <%= render partial: 'details_for_a_single_page', locals: {page_to_render: content_page, level: 1} %>
 
   <% end %>
 </div>
 <br>
-
-<%= link_to 'Add a new Page', new_content_page_path %>
```
Comment (marcrice12) [inline app/views/content_pages/index.html.erb:18]: i think some new pages will be required post launch

File: app/webpacker/styles/_cards.scss (hunk 2 above)
Comment (tonyto) [reply app/webpacker/styles/_cards.scss:8]: Also, why are classes prefixed with `nhsuk` - smells like copy and paste fail

File: app/models/content_page.rb (hunk 1 above)
Comment (guy-roberts) [reply app/models/content_page.rb:24]: Thats more concise,

But its harder to read the intention

File: app/views/layouts/_breadcrumbs.html.erb [hunk 4]
```diff
@@ -1,10 +1,10 @@
 <div class="govuk-breadcrumbs">
   <ol class="govuk-breadcrumbs__list">
     <li class="govuk-breadcrumbs__list-item">
-      <a class="govuk-breadcrumbs__link" href="https://www.gov.uk">Home</a>
+      <a class="govuk-breadcrumbs__link" href="<%= content_url %>">Home</a>
     </li>
     <li class="govuk-breadcrumbs__list-item">
-      <a class="govuk-breadcrumbs__link" href="#">Breadcrumbs</a>
+      <a class="govuk-breadcrumbs__link" href=#">Breadcrumbs will be here</a>
```
[the remark this answers is not in the document]
Comment (marcrice12) [reply app/views/layouts/_breadcrumbs.html.erb:7]: can you remove all changes to this file, as tunde and i are working on the breadcrumbs, hoping to complete soon

**Review by marcrice12 — COMMENTED** [review]
looks great

File: app/helpers/content_helper.rb [hunk 5]
```diff
@@ -1,2 +1,6 @@
 module ContentHelper
+  def subnav_link_css_classes(content_page)
```
[the remark this answers is not in the document]
Comment (marcrice12) [reply app/helpers/content_helper.rb:2]: do you mean use a shorter name?

[the remark this answers is not in the document]
Comment (marcrice12) [reply app/helpers/content_helper.rb:2]: what about just adding comments:
#passing in content_page object, returning css classes string for that object
#top level objects will not have parent_id

**Review by tonyto — APPROVED** [review]

**Pull request merged**